pycloudlib/cloud.py

```diff
@@ -224,6 +225,13 @@ def clean(self) -> List[Exception]:
                 self.delete_image(image_id)
             except Exception as e:
                 exceptions.append(e)
+        for image_info in self.preserved_images:
+            # send info log informing that the image is NOT being cleaned up
+            self._log.info(
+                "Preserved image %s [id:%s] is NOT being cleaned up.",
```

Comment:
LGTM. should be a conspicuous enough log message at the end of tests. thanks.
We'll need to update the `pycluldlib.Cloud.snapshot` signature and docstr to expect an optional `keep: bool = False` param too and reflect that functionality to each `pycloudlib.<could_name>.cloud.snapshot` if we could.

Reply:
Sounds good! Should I throw this back into draft state and make the changes across all clouds?

Reply:
no need to draft state this, I'll just await a request for re-review from you when you are ready on this.